File: src/main/java/pl/training/module7/breakdependencies/before/StandardMaintenanceWindows.java
package pl.training.module7.breakdependencies.before;

public final class StandardMaintenanceWindows {
    private static final int WINDOW_START_HOUR_UTC = 0;
    private static final int WINDOW_END_HOUR_UTC = 6;

    public boolean allows(String service, int hourUtc) {
        return hourUtc >= WINDOW_START_HOUR_UTC
                && hourUtc < WINDOW_END_HOUR_UTC;
    }
}

File: csharp/src/Training.Module7/BreakDependencies/Before/StandardMaintenanceWindows.cs
namespace Training.Module7.BreakDependencies.Before;

public sealed class StandardMaintenanceWindows
{
    private const int WindowStartHourUtc = 0;
    private const int WindowEndHourUtc = 6;

    public bool Allows(string service, int hourUtc) =>
        hourUtc >= WindowStartHourUtc
        && hourUtc < WindowEndHourUtc;
}
